File: src/main/java/com/uwetrottmann/trakt5/entities/TraktList.java
package com.uwetrottmann.trakt5.entities;

import com.uwetrottmann.trakt5.enums.ListPrivacy;
import com.uwetrottmann.trakt5.enums.SortBy;
import com.uwetrottmann.trakt5.enums.SortHow;
import org.threeten.bp.OffsetDateTime;

public class TraktList {

    public ListIds ids;
    public String name;
    public String description;
    public ListPrivacy privacy;
    public Boolean display_numbers;
    public Boolean allow_comments;
    public SortBy sort_by;
    public SortHow sort_how;
    public OffsetDateTime created_at;
    public OffsetDateTime updated_at;
    public Integer item_count;
    public Integer comment_count;
    public Integer likes;
    public User user;

    public TraktList id(ListIds id) {
        this.ids = id;
        return this;
    }

    public TraktList name(String name) {
        this.name = name;
        return this;
    }

    public TraktList description(String description) {
        this.description = description;
        return this;
    }

    public TraktList privacy(ListPrivacy privacy) {
        this.privacy = privacy;
        return this;
    }

    public TraktList displayNumbers(boolean displayNumbers) {
        this.display_numbers = displayNumbers;
        return this;
    }

    public TraktList allowComments(boolean allowComments) {
        this.allow_comments = allowComments;
        return this;
    }

    public TraktList sortBy(SortBy sortBy) {
        this.sort_by = sortBy;
        return this;
    }

    public TraktList sortHow(SortHow sortHow) {
        this.sort_how = sortHow;
        return this;
    }

}

File: src/main/kotlin/com/redissi/trakt/entities/TraktList.kt
package com.redissi.trakt.entities

import com.redissi.trakt.enums.ListPrivacy
import com.redissi.trakt.enums.SortBy
import com.redissi.trakt.enums.SortHow
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.time.OffsetDateTime

@JsonClass(generateAdapter = true)
class TraktList(
    val ids: ListIds? = null,
    val name: String? = null,
    val description: String? = null,
    val privacy: ListPrivacy? = null,
    @Json(name = "display_numbers")
    val displayNumbers: Boolean? = null,
    @Json(name = "allow_comments")
    val allowComments: Boolean? = null,
    @Json(name = "sort_by")
    val sortBy: SortBy? = null,
    @Json(name = "sort_how")
    val sortHow: SortHow? = null,
    @Json(name = "created_at")
    val createdAt: OffsetDateTime? = null,
    @Json(name = "updated_at")
    val updatedAt: OffsetDateTime? = null,
    @Json(name = "item_count")
    val itemCount: Int? = null,
    @Json(name = "comment_count")
    val commentCount: Int? = null,
    val likes: Int? = null,
    val user: User? = null
)